src/utils/__tests__/parser.test.ts

```diff
@@ -26,4 +26,42 @@ describe("parser", () => {
       expect(callWithBadParam).toThrow();
     });
   });
+
+  describe("parseCss()", () => {
+    test("throws when passed invalid css", () => {
+      const css = "color #fff";
+      expect(() => parseCss(css)).toThrow();
+    });
+
+    test("parses css declarations when classes, ids and elements are used", () => {
+      const css = `
+        div {
+          color: red;
+        }
+        #content {
+          width: 300px;
+        }
+        .main {
+          height: 100px;
+        }
+      `;
+      expect(parseCss(css)).toHaveLength(3);
+    });
+
+    test("returns array of PostCss AST declarations", () => {
+      const css = `
+        font-size: 10px;
+        color: #000;
+        font-weight: 700;
+        .icon {
+          width: 30px;
+        }
+      `;
+      const parsed = parseCss(css);
+      const matchDeclarations = parsed.every((item) => item.type === "decl");
+
+      expect(parsed instanceof Array).toBeTruthy();
+      expect(matchDeclarations).toBeTruthy();
+    });
+  });
```

Comment:
In my opinion, it's more convenient to test the parser/transformer together in an integration test, rather than unit testing the parser itself. That way the tests don't break when we refactor something (for example change the return type of the parser), as long as the parser and transformer work together.

I would only do some quick sanity checks here, like you already did with `toThrow`. I'd only add a small test for the happy path here, just to assert that it doesn't throw.

Since it's now possible to parse more complex rules with classes (nice!), I'd add a few tests under `css2js.test.js` to cover those cases.

Reply:
Thanks for the feedback, I'm fairly new to testing. I have added a test case in `css2js` to handle more complex rules, hopefully that's enough for now. Please have a look when you have a moment.

Reply:
@impulsivetomato thanks, that looks great!